### ISSUE TITLE
function MapFrom - ability to receive a dynamic value instead of constant string.

### ISSUE THREAD

**Issue by zhvateldima** [loedeman/AutoMapper#29]
Hi Bert, thank you for your lib, great job!

But I've got  a little issue with method"forMember", which receives a function as second parameter:
`mapper.forMember(propertyName1, function (opts) { opts.mapFrom("propertyName2"); } )`;

Method "forMember" ignores code that written like this:
`
let propertyName2 = gettingSomePropertyName;
mapper.forMember(propertyName1, function (opts) { opts.mapFrom(propertyName2); } )``

In other words, if propertyName2 does not have the same value as name of variable ("propertyName2") then mapper doesn't work. I've researched why and core of the problem is that callback function (second parameter of "forMember") is converted to the string and the parameter"propertyName2" extracted as it is (instead of taking a value).

Can you fix it or maybe can you suggest something?

Thank you.

Best regards,
Dima

**Comment by loedeman** [comment]
Hi Dima, thanks for using AutoMapperTS and locating this issue. Since a couple of months I am aware of this issue. Unfortunately, fixing this issue means quite some hassle, since the most obvious fixes significantly impact the library's performance. Since nobody complained until now, I decided to leave the issue in there until resolving it would be absolutely necessary. Are you able to work around this issue or are you currently unable to utilize the library? Cheers, Bert

**Comment by zhvateldima** [comment]
Thank you for a quick answer, Bert. And of course, it's not the big problem, I left some ugly workaround ("eval" - shame on me :D) for a while, but I hope problem will be gone :)

Dima

**Comment by loedeman** [comment]
I will look into it asap. I am currently quite busy at my day job, please be patient ;) . Of course, if you would like to solve the issue yourself, I will be more than happy to receive a PR :) ...
